base_phone_popup/models/phone_common.py

```diff
@@ -0,0 +1,122 @@
+            _logger.debug('searching on %s with domain=%s', obj._name, domain)
+            res_obj = obj.search(domain)
+            if len(res_obj) > 1:
+                _logger.warning(
+                    "There are several %s (IDS = %s) with a phone number "
+                    "ending with '%s'. Taking the first one.",
+                    obj._name, res_obj.ids, end_number_to_match)
+                res_obj = res_obj[0]
+            if res_obj:
+                name = res_obj.name_get()[0][1]
+                res = (obj._name, res_obj.id, name)
+                _logger.debug(
+                    "Answer get_record_from_phone_number: (%s, %d, %s)",
+                    res[0], res[1], res[2])
+                return res
+            else:
+                _logger.debug(
+                    "No match on %s for end of phone number '%s'",
+                    obj._name, end_number_to_match)
+        return False
+
+    @api.model
+    def _get_phone_models(self):
+        # Hook to get models and fields on which search
+        # You must inherit if you want to search into another fields
+        res = [
+            {'fields': ['phone', 'mobile'], 'object': self.env['res.partner']},
+            {'fields': ['phone', 'mobile'], 'object': self.env['crm.lead']},
+        ]
+        return res
+
+    @api.model
+    def _prepare_incall_pop_action(self, record_res, number):
+        if record_res:
+            obj = self.env[record_res[0]]
+            action = {
+                'name': obj._description,
+                'type': 'ir.actions.act_window',
+                'res_model': record_res[0],
+                'view_mode': 'form,tree',
+                'views': [[False, 'form']],  # Beurk, but needed
+                'target': 'new',
+                'res_id': record_res[1],
+                }
+        else:
+            action = {
+                'name': _('Number Not Found'),
+                'type': 'ir.actions.act_window',
+                'res_model': 'number.not.found',
+                'view_mode': 'form',
+                'views': [[False, 'form']],  # Beurk, but needed
+                'target': 'new',
+                'context': {'default_calling_number': number}
+            }
+        return action
+
+    @api.model
+    def incall_notify_by_login(self, number, login_list):
+        assert isinstance(login_list, list), 'login_list must be a list'
+        res = self.get_record_from_phone_number(number)
+        users = self.env['res.users'].search(
+            [('login', 'in', login_list)])
+        _logger.debug(
+            'Notify incoming call from number %s to users %s'
+            % (number, users.ids))
+        action = self._prepare_incall_pop_action(res, number)
+        if action:
+            for user in users:
+                if user.context_incall_popup:
+                    self.sudo(user.id).env['bus.bus'].sendone(
+                        '%s_%d' % (self._name, self.env.uid), action)
```

Comment:
@cubells  Did you test this PR in real life ? I'm not an expert of bus.bus, but I don't see how the above line of code could work. It seems you got inspiration from Anybox's module web_action_request on which base_phone_popup used to depend, and you took the code of the notify() method of that module, but you missed that this module has JS code that you didn't get here (and other small differences)...
On my side, I didn't succeed in making the above line of code work.

Reply:
I have web_notify in oca for that kind of things.

Reply:
Uhm, weird, this PR is supposed to use that module... (at least that were the requirements in the task). I haven't checked the PR deeply. @cubells can you clarify?

Reply:
I ported base_phone_popup to v10 using web_notify and I am very pleased with the result, cf this PR https://github.com/OCA/connector-telephony/pull/171/files  I think it's much better than the previous behavior which was too intrusive.
I think we should do the same for v11/v12. But, before that, someone needs to up-port the "interactive notification" feature of web_notify on which we rely (feature added by @guewen a few months ago, cf https://github.com/OCA/web/pull/970).